### PR TITLE
Frontend instrumentation

### PR DESCRIPTION
**Pull request by gtk-grafana**
We noticed our console.errors weren't magically making it into Faro when debugging https://github.com/grafana/explore-logs/pull/788, probably because we weren't sending them 🤦!

This PR attempts to add a runtime safe logging service to replace all console calls that will send frontend telemetry to Faro.

This was done quickly, but we should be asking "is the context in the log enough to adequately debug the problem?" for each log

### CODE REVIEW

File: src/services/variables.ts [hunk 1]
```diff
@@ -157,7 +158,7 @@ export function getValueFromFieldsFilter(filter: AdHocVariableFilter, variableNa
   try {
     return JSON.parse(filter.value);
   } catch (e) {
-    console.error(`Failed to parse ${variableName}`, e);
+    logger.error(e, { msg: `Failed to parse ${variableName}` });
```
Comment (gtk-grafana) [inline src/services/variables.ts:161]: should be a warning now that we've got a fallback

File: src/Components/ServiceScene/Breakdowns/Patterns/PatternsLogsSampleScene.tsx [hunk 2]
```diff
@@ -137,10 +138,10 @@ export class PatternsLogsSampleScene extends SceneObjectBase<PatternsLogsSampleS
       value.data.state === LoadingState.Error
     ) {
       // Logging an error so loki folks can debug why some patterns returned from the API seem to fail.
-      console.error('Pattern sample query returns no results', {
+      logger.error('Pattern sample query returns no results', {
         pattern: this.state.pattern,
-        traceIds: value.data.traceIds,
-        request: value.data.request,
+        traceIds: JSON.stringify(value.data.traceIds),
+        request: JSON.stringify(value.data.request),
```
Comment (gtk-grafana) [inline src/Components/ServiceScene/Breakdowns/Patterns/PatternsLogsSampleScene.tsx:144]: Could have a circular ref in here which would except

File: src/Components/ServiceScene/Breakdowns/PatternsLogsSampleScene.tsx [hunk 3]
```diff
@@ -136,10 +137,10 @@ export class PatternsLogsSampleScene extends SceneObjectBase<PatternsLogsSampleS
       value.data.state === LoadingState.Error
     ) {
       // Logging an error so loki folks can debug why some patterns returned from the API seem to fail.
-      console.error('Pattern sample query returns no results', {
+      logger.error('Pattern sample query returns no results', {
         pattern: this.state.pattern,
-        traceIds: value.data.traceIds,
-        request: value.data.request,
+        traceIds: JSON.stringify(value.data.traceIds),
+        request: JSON.stringify(value.data.request),
```
Comment (gtk-grafana) [inline src/Components/ServiceScene/Breakdowns/PatternsLogsSampleScene.tsx:143]: same here

File: src/Components/Table/ColumnSelection/ColumnSelectionDrawerWrap.tsx [hunk 4]
```diff
@@ -43,7 +44,9 @@ export function ColumnSelectionDrawerWrap() {
   const [searchValue, setSearchValue] = useState<string>('');
   const toggleColumn = (columnName: string) => {
     if (!columns || !(columnName in columns)) {
-      console.warn('failed to get column', columns);
+      logger.warn('failed to get column', {
+        columns: JSON.stringify(columns),
```
Comment (gtk-grafana) [inline src/Components/Table/ColumnSelection/ColumnSelectionDrawerWrap.tsx:48]: and here

File: src/services/logger.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  try {
+    if (err instanceof Error) {
+      logError(err, context);
+    } else if (typeof err === 'string') {
+      logError(new Error(err), context);
+    } else {
+      logError(new Error('unknown error'), context);
+    }
+  } catch (e) {
+    console.error('Failed to log faro error!', { err, context });
```
Comment (gtk-grafana) [inline src/services/logger.ts:45]: might be overkill, but I didn't want any runtime errors coming from failing to log something

**Review by svennergr — APPROVED** [review]
LGTM, I just added the plugin id as "default" ctx.

**Pull request merged**